src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs

```diff
@@ -123,11 +124,15 @@ public static class WebScriptHostBuilderExtension
                     services.TryAddSingleton<IJobHostMiddlewarePipeline, DefaultMiddlewarePipeline>();
                     services.TryAddEnumerable(ServiceDescriptor.Singleton<IJobHostHttpMiddleware, CustomHttpHeadersMiddleware>());
                     services.TryAddEnumerable(ServiceDescriptor.Singleton<IJobHostHttpMiddleware, HstsConfigurationMiddleware>());
-                    if (environment.IsAnyLinuxConsumption())
+
+                    if (environment.IsAnyLinuxConsumption() || environment.IsCorsConfigurationAllowed())
```

Comment:
you could store the result of `environment.IsAnyLinuxConsumption()` to a variable and use in line 128 and 134, so that the code inside that method gets executed only once.